### ISSUE TITLE
[Experiment] The default "Default Homepage Banner" is randomly displayed instead of the treatment's one after a Firefox restart

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla-mobile/firefox-ios#15046]
**[Affected versions]:**
- Firefox Release 114.2 (31625) build from TestFlight
- Firefox Release 113.2 (31081) build from TestFlight

**[Affected Platforms]:**
- iOS 15.6 - iPhone 13 mini
- iOS 16.2 - iPhone SE

**[Steps to reproduce]:**
1. Open the browser and tap the hamburger menu button from the bottom-right part of the screen.
2. Select the "Settings" option and scroll down to the bottom part of the menu.
3. Tap the "Firefox 114.2 (31625)" option 5 times.
4. Scroll down to the "Debug" section and select the "Experiments" option.
5. Tap the "Edit" button from the top-right part of the screen.
6. Tap on the "Use Preview Collection" radio button and restart the app.
7. Repeat steps from 1 to 5.
8. Tap the "Reset" button from the bottom part of the screen.
9. Tap the "Back" button from the top-left part of the page.
10. Tap the "Mobile default browser homepage banner copy test - iOS" experiment and select the "treatment-a" branch.
11. Restart the app and observe the "Default Homepage Banner".
12. Repeat step 11 a few times.

**[Expected results]:**
- The same "Default Homepage Banner" message specific to treatment-a is displayed after each Firefox restart.

**[Actual results]:**
- Either the default "Default Homepage Banner" or the message specific to treatment-a is displayed after restarting Firefox.

**[Notes]**:
- Here is a screen recording of the issue: [Gdoc](https://drive.google.com/file/d/1sn2J1cvIxYESFLCPhVIra3o_rxeEaLfr/view?usp=sharing).

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-6761)

**Comment by nishant2718** [comment]
Hello! I checked the experiment setup on https://experimenter.services.mozilla.com/nimbus/mobile-default-browser-cta-copy-test-ios/summary and it looks like the treatment has the same priority at the control, which is FALLBACK. 

If the priority is the same, there's no guarantee of which message surfaces first between two messages of the same priority. 

The control / evergreen needs to always have the least priority (FALLBACK) and the treatments need to always be above that. 

To confirm this, could you please check with a treatment that has a priority above FALLBACK? See this for reference on priority of messages, https://github.com/mozilla-mobile/firefox-ios/blob/ba45c54d32ba2ca82df1480af9afb8faa4eed61a/nimbus-features/messagingFeature.yaml#L93.

**Comment by SoftVision-CosminMuntean** [comment]
Hi! I'm sorry for my late reply. I have created a [new Experiment](https://stage.experimenter.nonprod.dataops.mozgcp.net/nimbus/cmuntean-verify-ios-experiment/summary) as Nishant suggested in the above comment with  "style": "PERSISTENT" for the treatment branch. 
- I have verified this on the latest Firefox 118.0 build created from the main branch using the new experiment. After enrolling in the treatment branch, the message specific to treatment-a remains displayed after restarting Firefox.

Considering this I will close this issue as verified.
